core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -56,7 +55,7 @@ private[spark] object JsonProtocol {
 
   private implicit val format = DefaultFormats
 
-  private val mapper = new ObjectMapper().registerModule(DefaultScalaModule)
```

Comment:
This was added in #10061 in order to allow the new `SparkListenerSQLExecutionStart` and `SparkListenerSQLExecutionEnd` events to be written to the event log using Jackson. I'll see if there's an existing unit test for roundtrip serialization of these events.